[CONTP-1795] Add agent performance metrics to agent telemetry page (#38045)
DataDog/documentation: content/en/data_security/agent.md

| workloadmeta.pull_errors                    | Number of WorkloadMeta pull errors                                                                                     |
| appsec_injector.watched_changes             | Number of changes detected by the AppSec injector for watched resources                                                |
| appsec_injector.sidecar_mutations           | Number of AppSec injector sidecar admission outcomes (pod mutation and deletion)                                       |

> | agent_performance.containers_restarts       | Sum of container restart counts for Datadog Cluster Agent and Cluster Checks Runner pods                              |

Reviewer: ```suggestion
| agent_performance.containers_restarts       | Number of container restarts for the Cluster Agent and Cluster Checks Runner pods                              |
```

Reviewer: Taking another look, I've realized that the other entries in this table all use `Number of` or `Total` for similar descriptions. Suggesting to update this for consistency.

Also suggesting to drop:
- `counts` for redundancy
- `Datadog` as being unnecessary given the context (the table section title itself is just "Cluster Agent"

> | agent_performance.containers_terminated     | Sum of container terminations for Datadog Cluster Agent and Cluster Checks Runner pods, tagged by reason               |

Reviewer: ```suggestion
| agent_performance.containers_terminated     | Number of container terminations for Cluster Agent and Cluster Checks Runner pods, tagged by reason               |
```

| agent_performance.memory_usage              | Sum of container runtime memory usage for Datadog Cluster Agent and Cluster Checks Runner pods                        |

> | agent_performance.memory_limit              | Sum of container runtime memory limits for Datadog Cluster Agent and Cluster Checks Runner pods                       |

Reviewer: "Sum of" or "count of" for these metrics is a bit confusing in the absence of a unit to measure by. For example, the `pymem.inuse` metric on line 156 in this table uses `byte`:
> Total container runtime memory usage, in bytes, for the Datadog Cluster Agent and Cluster Checks Runner pods

Would it be possible to update these descriptions accordingly, with whatever units are applicable for these metrics?

| workloadmeta.pull_errors                    | Number of WorkloadMeta pull errors                                                                                     |
| appsec_injector.watched_changes             | Number of changes detected by the AppSec injector for watched resources                                                |
| appsec_injector.sidecar_mutations           | Number of AppSec injector sidecar admission outcomes (pod mutation and deletion)                                       |
| **Agent Performance**                       |                                                                                                                        |
| kubernetes_agent.containers_restarts        | Sum of container restart counts for Datadog Cluster Agent and Cluster Checks Runner pods                              |

> | kubernetes_agent.containers_terminated      | Sum of kubelet container terminated-state metrics for Datadog Cluster Agent and Cluster Checks Runner pods, preserving the `reason` tag |

Reviewer: ```suggestion
| kubernetes_agent.containers_terminated      | Sum of container terminations for Datadog Cluster Agent and Cluster Checks Runner pods, tagged by reason` |
```

Reviewer: Suggestion to follow the established pattern for describing a tag dimension on a metric — see line 171 ("Count of HTTP connectivity retry attempts, tagged by status (success or failure)"). "Preserving the `reason` tag" doesn't match that convention and is less clear about what the tag represents.

Author: That makes sense. I'll make the change.

| workloadmeta.pull_errors                    | Number of WorkloadMeta pull errors                                                                                     |
| appsec_injector.watched_changes             | Number of changes detected by the AppSec injector for watched resources                                                |
| appsec_injector.sidecar_mutations           | Number of AppSec injector sidecar admission outcomes (pod mutation and deletion)                                       |
| agent_performance.containers_restarts       | Number of container restarts for the Cluster Agent and Cluster Checks Runner pods                                      |
| agent_performance.containers_terminated     | Number of container terminations for Cluster Agent and Cluster Checks Runner pods, tagged by reason                    |
| agent_performance.memory_usage              | Total container runtime memory usage, in bytes, for Cluster Agent and Cluster Checks Runner pods                       |

> | agent_performance.memory_limit              | Total container runtime memory limits, in bytes, for Cluster Agent and Cluster Checks Runner pods                      |

Reviewer: ```suggestion
| agent_performance.containers_terminated     | Number of container terminations for the Cluster Agent and Cluster Checks Runner pods, tagged by reason                    |
| agent_performance.memory_usage              | Total container runtime memory usage, in bytes, for the Cluster Agent and Cluster Checks Runner pods                       |
| agent_performance.memory_limit              | Total container runtime memory limits, in bytes, for the Cluster Agent and Cluster Checks Runner pods                      |
```
